runtime/src/builtin/shared.rs

```diff
@@ -25,17 +26,19 @@ where
 {
     // if we are able to resolve it to an ID address, return the resolved address
     if let Some(addr) = rt.resolve_address(address) {
-        return Ok(addr);
+        return Ok(Address::new_id(addr));
```

Comment:
Could we change this to return ActorID? Would that cause lots of noise at callers?